lib/experimental/sync/class-gutenberg-http-signaling-server.php

```diff
@@ -0,0 +1,380 @@
+<?php
+/**
+ * Gutenberg_HTTP_Signaling_Server class
+ *
+ * @package    Gutenberg
+ */
+
+/**
+ * Gutenberg class that contains an HTTP based signaling server used for collaborative editing.
+ *
+ * @access private
+ * @internal
+ */
+class Gutenberg_HTTP_Signaling_Server {
+
+
+	/**
+	 * Adds a wp_ajax action to handle the signaling server requests.
+	 */
+	public static function init() {
+		$gutenberg_experiments = get_option( 'gutenberg-experiments' );
+		if ( ! $gutenberg_experiments || ! array_key_exists( 'gutenberg-sync-collaboration', $gutenberg_experiments ) ) {
+			return;
+		}
+		add_action( 'wp_ajax_gutenberg_signaling_server', array( __CLASS__, 'do_wp_ajax_action' ) );
+	}
+
+	/**
+	 * Handles a wp_ajax signaling server request.
+	 */
+	public static function do_wp_ajax_action() {
+		if ( empty( $_REQUEST ) || empty( $_REQUEST['subscriber_id'] ) ) {
+			die( 'no identifier' );
```

Comment:
Now that we have errors, what do you think of sending this as messages back to the client instead of silently dying? There's a few in this method.